lib/factory_trace/structures/factory.rb

```diff
@@ -34,7 +36,9 @@ def merge!(factory)
       def ==(factory)
         return false unless factory.is_a?(FactoryTrace::Structures::Factory)
 
-        name == factory.name && parent_name == factory.parent_name && trait_names == factory.trait_names
+        (name == factory.name || (alias_names & factory.alias_names).present?) &&
```

Comment:
BTW, `alias_names & factory.alias_names` it should not be just present it should be equal.
Your condition succeed for `[1] & [1,2]` but should not. 

Please just do `alias_names == factory.alias_names`